packages/bs-solana/src/BSSolana.ts

```diff
@@ -0,0 +1,310 @@
+
+export class BSSolana<BSName extends string = string>
+  implements
+    BlockchainService<BSName, any>,
+    BSCalculableFee<BSName>,
+    BSWithNameService,
+    BSWithLedger<BSName>,
+    BSWithNft,
+    BSWithExplorerService
+{
+  name: BSName
+  bip44DerivationPath: string
+
+  feeToken!: Token
+  tokens!: Token[]
+  network!: Network<BSSolanaNetworkId>
+
+  ledgerService: Web3LedgerServiceSolana<BSName>
+  exchangeDataService!: ExchangeDataService
+  blockchainDataService!: BlockchainDataService
+  nftDataService!: NftDataService
+  explorerService!: ExplorerService
+
+  #connection!: solanaSDK.Connection
+  #apiKeys: BSSolanaApiKeys
+
+  constructor(
+    name: BSName,
+    apiKeys: BSSolanaApiKeys,
+    network?: Network<BSSolanaNetworkId>,
+    getLedgerTransport?: GetLedgerTransport<BSName>
+  ) {
+    network = network ?? BSSolanaConstants.DEFAULT_NETWORK
+
+    this.name = name
+    this.bip44DerivationPath = BSSolanaConstants.DEFAULT_BIP44_DERIVATION_PATH
+    this.ledgerService = new Web3LedgerServiceSolana(this, getLedgerTransport)
+
+    this.#apiKeys = apiKeys
+
+    this.setNetwork(network)
+  }
+
+  #generateKeyPairFromKey(key: string): solanaSDK.Keypair {
+    let keyBuffer: Uint8Array | undefined
+
+    try {
+      keyBuffer = bs58.decode(key)
+    } catch {
+      keyBuffer = Uint8Array.from(key.split(',').map(Number))
+    }
+
+    return solanaSDK.Keypair.fromSecretKey(keyBuffer)
+  }
+
+  async #signTransaction(transaction: solanaSDK.Transaction, senderAccount: Account<BSName>) {
+    if (senderAccount.isHardware) {
+      if (!this.ledgerService.getLedgerTransport)
+        throw new Error('You must provide getLedgerTransport function to use Ledger')
+
+      if (typeof senderAccount.bip44Path !== 'string')
+        throw new Error('Your account must have bip44 path to use Ledger')
+
+      const transport = await this.ledgerService.getLedgerTransport(senderAccount)
+      return await this.ledgerService.signTransaction(transport, transaction, senderAccount)
+    }
+
+    transaction.sign(this.#generateKeyPairFromKey(senderAccount.key))
+    return transaction.serialize()
+  }
+
+  async #buildTransferParams(param: TransferParam) {
+    const latestBlockhash = await this.#connection.getLatestBlockhash()
+
+    const senderPublicKey = new solanaSDK.PublicKey(param.senderAccount.address)
+
+    const transaction = new solanaSDK.Transaction()
+    transaction.feePayer = senderPublicKey
+    transaction.recentBlockhash = latestBlockhash.blockhash
+    transaction.lastValidBlockHeight = latestBlockhash.lastValidBlockHeight
+
+    for (const intent of param.intents) {
+      const amount = Number(parseTokenNumber(intent.amount, intent.tokenDecimals))
+      const receiverPublicKey = new solanaSDK.PublicKey(intent.receiverAddress)
+      const normalizedTokenHash = normalizeHash(intent.tokenHash, false)
+
+      const isNative = normalizedTokenHash === normalizeHash(BSSolanaConstants.NATIVE_TOKEN.hash, false)
```

Comment:
Why didn't you use a lowercase here?

Reply:
Hashes has uppercases letters.
0x0D23 is different of 0x0d23